### ISSUE TITLE
NormalizeWhitespace breaks XML attributes in comments

### ISSUE THREAD

**Issue by svick** [dotnet/roslyn#47363]
**Version Used**: 3.7.0

**Steps to Reproduce**:

Run the following code:

```c#
string code = @"
/// <example>
///   <code lang=""cs"" source=""foo.cs"" />
/// </example>
class Foo {}";

var tree = SyntaxFactory.ParseSyntaxTree(code);
tree = tree.WithRootAndOptions(tree.GetRoot().NormalizeWhitespace(), tree.Options.WithDocumentationMode(DocumentationMode.Diagnose));
var compilation = CSharpCompilation.Create(null)
	.WithOptions(new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
	.AddSyntaxTrees(tree)
	.AddReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location));
var diagnostics = compilation.GetDiagnostics();
```

**Expected Behavior**:

No warnings.

**Actual Behavior**:

> warning CS1570: XML comment has badly formed XML -- ''source' is an unexpected token. Expecting whitespace. Line 2, position 32.'

This seems to happen because `NormalizeWhitespace()` changes the code to:

```c#
/// <example>
///   <code lang = "cs"source = "foo.cs"/>
/// </example>
class Foo
{
}
```

Notice how there is no space before the `source` attribute, which is invalid XML.

Also, when running this code under a Debug build of the compiler on .Net Core, the following assertion is triggered:

https://github.com/dotnet/roslyn/blob/4eca967f780e307b098e9d76d82e851983e7023f/src/Compilers/CSharp/Portable/Compiler/DocumentationCommentCompiler.cs#L1233-L1234

**Comment by Youssef1313** [comment]
*Part* of this is being fixed in https://github.com/dotnet/roslyn/pull/51981. I may look to fully fix this in a follow-up. Will need to review all the cases where EndQuoteToken can appear to decide when and when not to have a space after it.
